search/indexing_api.py

```diff
@@ -248,6 +248,7 @@ def program_enrolled_user_mapping():
     mapping.field("program", "object", properties={
         'id': LONG_TYPE,
         'grade_average': LONG_TYPE,
+        'non_learner': BOOL_TYPE,
```

Comment:
why don't you call this `is_learner` and reverse the logic so that you include in the result only learners instead of excluding the other roles?